File: Service/Admin/RMA.Service.Admin.MasterDataManager.Contracts/Entities/AuthorityLimitRequest.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.Admin.MasterDataManager.Contracts.Entities
{
    public class AuthorityLimitRequest
    {
        public AuthorityLimitItemTypeEnum AuthorityLimitItemType { get; set; }
        public AuthorityLimitValueTypeEnum? AuthorityLimitValueType { get; set; }
        public AuthorityLimitContextTypeEnum? AuthorityLimitContextType { get; set; } // for context under what to be tracked
        public int? ContextId { get; set; } // the contexts id
        public int Value { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/shared-models-lib/src/lib/authority-limits/authority-limit-request.ts
import { AuthorityLimitContextTypeEnum } from "../enums/authority-limits/authority-limit-context-type-enum";
import { AuthorityLimitItemTypeEnum } from "../enums/authority-limits/authority-limit-item-type-enum";
import { AuthorityLimitValueTypeEnum } from "../enums/authority-limits/authority-limit-value-type-enum";

export class AuthorityLimitRequest {
    authorityLimitItemType: AuthorityLimitItemTypeEnum;
    authorityLimitValueType: AuthorityLimitValueTypeEnum;
    authorityLimitContextType: AuthorityLimitContextTypeEnum;
    contextId: number;
    value: number;
}